STYLE_GUIDE.md

```diff
@@ -0,0 +1,174 @@
+# Composer Style Guide
+
+This guide serves as a rule-of-thumb for how code should be organized and formatted in Composer.
+For general python styling questions not covered in this document, consult Google's
+[Python Style Guide](https://google.github.io/styleguide/pyguide.html).
+
+## 1. Code Formatting
+
+Composer uses the [yapf](https://github.com/google/yapf) formatter for general styling and
+[isort](https://github.com/PyCQA/isort) to sort imports.
+
+To format code, run
+
+```bash
+yapf -rip .  # for yapf
+isort .  # for isort
+```
+
+The configuration is stored in [pyproject.toml](pyproject.toml).
+
+
+## 2. Type Annotations and Typechecking
+
+Composer aims to annotate all functions with type annotations (introduced in
+[PEP 526](https://www.python.org/dev/peps/pep-0526/). Type annotations help statically catch `TypeError` and
+`AttributeError` bugs, in addition to other benefits, as outlined in the PEP.
+
+Composer uses [pyright](https://github.com/microsoft/pyright)
+to validate type annotations. To check typing, run:
+
+```bash
+pyright .
+```
+
+Our pyright configuration is stored in [pyproject.toml](pyproject.toml).
+
+While pyright warnings will not cause a build to fail, they should be reviewed as they can identify potential bugs.
+
+### Debugging
+
+Here are some suggestions to deal with pyright errors:
```

Comment:
It would be nice to have an example of using `assert` for type checking here. :)